src/local_scheduler/local_scheduler.cc

```diff
@@ -572,7 +581,12 @@ void assign_task_to_worker(LocalSchedulerState *state,
   worker->task_in_progress = Task_copy(task);
   /* Update the global task table. */
   if (state->db != NULL) {
+#if !RAY_USE_NEW_GCS
     task_table_update(state->db, task, NULL, NULL, NULL);
+#else
+    RAY_CHECK_OK(TaskTableAdd(&state->gcs_client, task));
+    Task_free(task);
```

Comment:
so the semantics have changed, is that right? Whereas `task_table_update` takes ownership of `task`, `TaskTableAdd` does not, is that right?

Reply:
Yes